### PR TITLE
Osms

### PR DESCRIPTION
**Pull request by AhmedRaihanAlif**

### CODE REVIEW

File: Controllers/AdminPanel/addHrLocationsAll.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    lastUpdateDate: Joi.string().min(0),
+    lastUpdatedBy: Joi.number().max(15),
+    lastUpdateLogin: Joi.number().max(15),
+    createdBy: Joi.number().max(15),
+    creationDate: Joi.string().min(0),
+  });
+  const validation = schema.validate(req.body);
+  if (validation.error) {
+    res.status(400).send("Invalid inputs");
+    next();
```
Comment (Remark2023) [inline Controllers/AdminPanel/addHrLocationsAll.js:33]: remove this line

File: Controllers/AdminPanel/deleteHrLocations.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const pool = require("/Project Work/OSMS/osms-backend/dbConnection");
+const router = express.Router();
+
+router.delete("/:location_id", async (req, res, next) => {
+  const locationId = req.params.location_id;
+  await pool.query(
+    "DELETE FROM hr_locations_all WHERE location_id = $1",
```
Comment (Remark2023) [inline Controllers/AdminPanel/deleteHrLocations.js:8]: sql will be for update

File: Controllers/AdminPanel/deleteHrLocations.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const pool = require("/Project Work/OSMS/osms-backend/dbConnection");
+const router = express.Router();
+
+router.delete("/:location_id", async (req, res, next) => {
```
Comment (Remark2023) [inline Controllers/AdminPanel/deleteHrLocations.js:5]: method will be put

File: index.js [hunk 4]
```diff
@@ -3,33 +3,39 @@ const express = require("express");
...
-const CompareOtpService = require("./controllers/compareOtpService");
+const CompareOtpService = require("./Controllers/compareOtpService");
 const StoreOtpService = require("../osms-backend/Controllers/storeOtpService");
 const GetOtpService = require("./Controllers/getOtpService");
 const DeleteOtpService = require("./Controllers/deleteOtpService");
+
+//routing api for admin
+const AddHrLocationsAll = require("./Controllers/AdminPanel/addHrLocationsAll");
 const AddUnitMeasureService = require("./Controllers/AdminPanel/addUnitMeasuresService");
-const GetUnitMeasureService = require("./Controllers/AdminPanel/getUnitMeasureService");
```
Comment (Remark2023) [inline index.js:14]: add these line

const GetUnitMeasureService = require("./Controllers/AdminPanel/getUnitMeasureService");
const DisableUnitMeasureService = require("./Controllers/AdminPanel/disableUnitMeasureService");
const UpdateUnitMeasureService = require("./Controllers/AdminPanel/updateUnitMeasureService");

File: index.js [hunk 5]
```diff
@@ -3,33 +3,39 @@ const express = require("express");
...
 app.use("/signup", SignupService);
 app.use("/send-otp", SendOtpService);
 app.use("/compare-otp", CompareOtpService);
 app.use("/store-otp", StoreOtpService);
 app.use("/get-otp", GetOtpService);
 app.use("/delete-otp", DeleteOtpService);
+
+// routing middleware for admin
 app.use("/add-unit-measure", AddUnitMeasureService);
-app.use("/get-unit-measure", GetUnitMeasureService);
```
Comment (Remark2023) [inline index.js:30]: add these lines

app.use("/get-unit-measure", GetUnitMeasureService);
app.use("/disable-unit-measure", DisableUnitMeasureService);
app.use("/update-unit-measure", UpdateUnitMeasureService);

**Review by Remark2023 — CHANGES_REQUESTED** [review]

**Comment by AhmedRaihanAlif** [comment]
Done all the conflict

**Pull request merged**